src/ext/Bal/wixext/BalErrors.cs

```diff
@@ -43,6 +43,11 @@ public static Message IuibaPrimaryPackageEnableFeatureSelection(SourceLineNumber
             return Message(sourceLineNumbers, Ids.IuibaPrimaryPackageEnableFeatureSelection, "When using WixInternalUIBootstrapperApplication, primary packages must not have feature selection enabled because it interferes with the user selecting feature through the MSI UI.");
         }
 
+        public static Message MissingDNCBAFactoryAssembly(SourceLineNumber sourceLineNumbers)
+        {
+            return Message(sourceLineNumbers, Ids.MissingDNCBAFactoryAssembly, "The BA's entry point DLL must have bal:BAFactoryAssembly=\"yes\" when using the DotNetCoreBootstrapperApplicationHost.");
```

Comment:
It would be better to use element names in the error messages to help direct users what to look for next. For example, instead of `BA` use `BootstrapperApplication` element and instead of just `entry point DLL` it would be good to mention the `Payload` element.